packages-exp/auth-exp/src/api/index.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Auth } from '../model/auth';
+import {
+  JsonError,
+  SERVER_ERROR_MAP,
+  ServerErrorMap,
+  ServerError
+} from './errors';
+import { AUTH_ERROR_FACTORY, AuthErrorCode } from '../core/errors';
+import { FirebaseError } from '@firebase/util';
+import { IdTokenResponse } from '../model/id_token';
+
+export enum HttpMethod {
+  POST = 'POST',
+  GET = 'GET'
+}
+
+export enum Endpoint {
+  CREATE_AUTH_URI = '/v1/accounts:createAuthUri',
+  DELETE_ACCOUNT = '/v1/accounts:delete',
+  RESET_PASSWORD = '/v1/accounts:resetPassword',
+  SIGN_UP = '/v1/accounts:signUp',
+  SIGN_IN_WITH_CUSTOM_TOKEN = '/v1/accounts:signInWithCustomToken',
+  SIGN_IN_WITH_EMAIL_LINK = '/v1/accounts:signInWithEmailLink',
+  SIGN_IN_WITH_IDP = '/v1/accounts:signInWithIdp',
+  SIGN_IN_WITH_PASSWORD = '/v1/accounts:signInWithPassword',
+  SIGN_IN_WITH_PHONE_NUMBER = '/v1/accounts:signInWithPhoneNumber',
+  SEND_VERIFICATION_CODE = '/v1/accounts:sendVerificationCode',
+  SEND_OOB_CODE = '/v1/accounts:sendOobCode',
+  SET_ACCOUNT_INFO = '/v1/accounts:update',
+  GET_ACCOUNT_INFO = '/v1/accounts:lookup',
+  GET_RECAPTCHA_PARAM = '/v1/recaptchaParams',
+  START_PHONE_MFA_ENROLLMENT = '/v2/accounts/mfaEnrollment:start',
+  FINALIZE_PHONE_MFA_ENROLLMENT = '/v2/accounts/mfaEnrollment:finalize',
+  START_PHONE_MFA_SIGN_IN = '/v2/accounts/mfaSignIn:start',
+  FINALIZE_PHONE_MFA_SIGN_IN = '/v2/accounts/mfaSignIn:finalize',
+  WITHDRAW_MFA = '/v2/accounts/mfaEnrollment:withdraw'
+}
+
+export async function performApiRequest<T, V>(
+  auth: Auth,
+  method: HttpMethod,
+  path: Endpoint,
+  request?: T,
+  customErrorMap: Partial<ServerErrorMap<ServerError>> = {}
+): Promise<V> {
+  const errorMap = { ...SERVER_ERROR_MAP, ...customErrorMap };
+  try {
+    const body = request
+      ? {
+          body: JSON.stringify(request)
+        }
+      : {};
+    const response = await fetch(
```

Comment:
You seem to hardcode the `fetch` call. I suppose you can polyfill this. But not sure this can be polyfilled for all environments (eg. Node.js, react-native). What is your plan regarding this.

Reply:
We are investigating react-native right now, suppose this can be passed in from the Auth object like we do with popupRedirectResolver